server/pubsub/redis_query_results.go

```diff
@@ -112,6 +113,11 @@ func (r *redisQueryResults) ReadChannel(ctx context.Context, query fleet.Distrib
 
 	// pub-sub can publish and listen on any node in the cluster
 	conn := redis.ReadOnlyConn(r.pool, r.pool.Get())
+	if err := conn.Err(); err != nil {
+		logging.WithErr(ctx, err)
+		logging.WithExtras(ctx, "pubsub", "get connection from pool failed")
```

Comment:
So, I'm not even sure this would log, as the results are returned in a websocket connection IIRC and it may not get to the logRequestEnd part of the web request that actually logs the logging context? But I'm not sure how else I can get a logger here?

Reply:
I added a call to log with the service's logger here, I think it should work? https://github.com/fleetdm/fleet/pull/3362/files#diff-15b7dd0f9bf87990467874aa516bbf2a1c8371f17490836bb5e78da0a2f568c4